build-tools/packages/build-tools/src/common/npmPackage.ts

```diff
@@ -490,11 +489,28 @@ export function updatePackageJsonFile(
 	packageTransformer: (json: PackageJson) => void,
 ): void {
 	const packagePath = path.join(packageDir, "package.json");
-	const indentation = detectIndent(packagePath).indent || "\t";
-	const pkgJson: PackageJson = readJsonSync(packagePath);
+	const [pkgJson, indent] = readPackageJsonAndIndent(packagePath);
 
 	// Transform the package.json
 	packageTransformer(pkgJson);
 
-	writeJsonSync(packagePath, sortPackageJson(pkgJson), { spaces: indentation });
+	writePackageJson(packagePath, pkgJson, indent);
+}
+
+/**
+ * Reads a package.json file from a path, detects its indentation, and returns both the JSON as an object and
+ * indentation.
+ */
+function readPackageJsonAndIndent(pathToJson: string): [json: PackageJson, indent: string] {
+	const contents = readFileSync(pathToJson).toString();
+	const indentation = detectIndent(contents).indent || "\t";
+	const pkgJson: PackageJson = JSON.parse(contents);
+	return [pkgJson, indentation];
+}
+
+/**
+ * Writes a PackageJson object to a file using the provided indentation.
+ */
+async function writePackageJson(packagePath: string, pkgJson: PackageJson, indent: string) {
+	writeJson(packagePath, sortPackageJson(pkgJson), { spaces: indent });
```

Comment:
dangling promise here. Either make this sync (ex: use writeJsonSync), or actually properly await it.

Its concerning if our lint setup doesn't catch this, as this is a common think to link for and can cause really nasty race condition bugs.

Reply:
The lint setup for @fluidframework/build-tools is haphazard. There was none when I started, and our shared configs were
tough to onboard because they're appropriately strict, so I extended the eslint/typescript-eslint recommended configs. There's also a lot of unused code, so we put something on the
backlog to clean it up once we'd figured out what needed to be preserved from the old code. We're at that point now,
though, so I think we can schedule that work.

For this PR I'll manually enable the promise-related lint rules.

Reply:
There are more offenses than I want to fix in this PR, so I'll fix the ones in this file and update the lint config in a
separate PR.

Reply:
I ended up just making this synchronous.